aips/aip-x-safe-auth-key-mapping.md

```diff
@@ -0,0 +1,221 @@
+   authenticated by the same authentication key: the original account whose
+   address is its authentication key, and another account that has had its
+   authentication key rotated to the authentication key of the original account.
+   Since `OriginatingAddress` is one-to-one, a dual-account situation can
+   inhibit indexing and OpSec (resolved in this PR with
+   `set_originating_address` private entry function).
+
+# Impact
+
+1. Without the changes proposed in this AIP's reference implementation,
+   unproven authentications (specifically those relying on
+   `rotate_authentication_key_call`) will result in an unidentifiable mapping,
+   such that users will be unable to identify accounts secured by their private
+   key unless they have maintained their own offchain mapping. This applies to
+   exotic wallets like passkeys.
+1. The overwrite behavior (described above) for
+   `update_auth_key_and_originating_address_table` can similarly result in an
+   inability to identify an account based on the private key.
+1. A user who authenticates two accounts with the same private key per the above
+   schema will experience undefined behavior during indexing and OpSec due to
+   the original one-to-one mapping assumption.
+
+# Summary
+
+The onchain key rotation address mapping has functional issues which inhibit
+safe mapping of authentication key to originating address for standard accounts.
+I propose resolving these issues with the reference implementation.
+
+# Proposed solution
+
+Assorted checks and extra function logic in [`aptos-core` #14309]
+
+# Alternative solutions
+
+Per @davidiw in a separate chat:
+
+> something where we update the rotation command to take in new and removed keys
+> as well as any additional metadata that might be useful such as parameters for
+> the authkey type. Then we add to the indexer a set of all keys that point to
+> potential addresses.
+
+As I understand, this approach would require breaking changes and would
+introduce offchain indexing as an additional dependency in the authentication
+key mapping paradigm.
+
+My solution, captured in the proposed reference implementation, offers a
+purely onchain solution to existing issues and does not require altering the
+existing design space or introducing an offchain dependency.
+
+# Specification
+
+N/A
+
+# Reference implementations
+
+[`aptos-core` #14309]
+
+# Risks and drawbacks
+
+Enforces a one-to-one mapping of private key to account address in the general
+case of following best practices, which extreme users (wishing to use one
+private key to authenticate all their accounts) may find restrictive.
+
+# Security considerations
+
+Note that the function `account::set_originating_address` proposed in
+[`aptos-core` #14309] must remain a private entry function to prevent unproven
+key rotation attacks.
+
+# Multisig considerations
+
+In a separate chat, @davidiw asked about how the changes in the reference
```

Comment:
I would remove the reference to David's question and just focus on the multisig consideration itself.

Reply:
Addressed in 4e5fb6a